### ISSUE TITLE
Adapt linux and macOS deployment infrastructure for py3

### ISSUE THREAD

**Issue by palango** [raiden-network/raiden#1205]
## Problem Definition

The deployment tools for linux and macOS haven't been tested under py3 and updated `ethereum` library.

## Solution

Test deployment tools

## Tasklist

- [x] Test linux deployment
- [x] Test macOS deployment

**Comment by palango** [comment]
@konradkonrad I remember there were soem encoding issues with the pyinstaller build on linux as well. Could you please test with master again.

**Comment by palango** [comment]
I need some help from all linux users to test the pyinstaller package on linux.

Can you please run the following commands on the latest raiden master branch. 
```bash
python setup.py compile_contracts
python setup.py compile_webui
pip install pyinstaller
pip uninstall -y pdbpp
pyinstaller --noconfirm --clean raiden.spec
```

Afterwards you will find a executable called `raiden-0.2.0-[githash]-linux` in the `dist` folder.

Please run this in your normal shell and another one where the `LC_ALL` and `LANG` environment vars are not set. If running `raiden smoketest` or `raiden` produces any warnings please post them here.

Afterwards don't forget to reinstall `pdbpp`.

**Comment by konradkonrad** [comment]
```
docker run --rm -it --entrypoint /bin/bash -v $(pwd)/out:/out python:3.6
# now inside container
apt-get update && apt-get install -y git-core wget && git clone https://github.com/raiden-network/raiden

cd raiden && pip install -r requirements.txt

wget -O /usr/bin/solc https://github.com/ethereum/solidity/releases/download/v0.4.18/solc-static-linux && chmod +x /usr/bin/solc

wget -O /tmp/geth.tar.gz https://gethstore.blob.core.windows.net/builds/geth-linux-amd64-1.7.3-4bb3c89d.tar.gz && cd /tmp && tar xzvf geth.tar.gz && mv geth-linux-amd64-1.7.3-4bb3c89d/geth /usr/bin/geth && cd -

wget -O /tmp/node.tar.gz https://nodejs.org/download/release/v8.2.1/node-v8.2.1-linux-x64.tar.gz && cd /tmp && tar xzvf node.tar.gz && mkdir /tmp/node_modules && chmod -R a+rwX /tmp/node_modules && cd -

python setup.py build

USER=root NPM_CONFIG_PREFIX=/tmp/node_modules NODE_PATH=/tmp/node_modules PATH=/tmp/node-v8.2.1-linux-x64/bin:$PATH python setup.py compile_webui

pip install .

pip install pyinstaller

pyinstaller --noconfirm --clean raiden.spec

cp dist/raiden-0.2.0+git.r4803554-linux /out/

exit
# now outside of container
out/raiden-0.2.0+git.r4803554-linux smoketest

>>> [1/5] getting smoketest configuration
>>> [2/5] starting ethereum
>>> [3/5] starting raiden
>>> [4/5] running smoketests...
>>> [5/5] smoketest successful, report was written to /tmp/tmphl1_dz4r.log

```
warnings
```
# during compile_webui
@angular/compiler-cli@5.2.4 requires typescript@'>=2.4.2 <2.7.0' but 2.7.1 was found instead.
Using this version can result in undefined behaviour and difficult to debug problems.
Please run the following command to install a compatible version of TypeScript.
    npm install typescript@'>=2.4.2 <2.7.0'
To disable this warning run "ng set warnings.typescriptMismatch=false".
```

**Comment by LefterisJP** [comment]
I got an error when running with an unset `LANG` variable.

```
$ PATH="/home/lefteris/w/go/src/github.com/ethereum/go-ethereum/build/bin/:/home/lefteris/ew/solidity/build/solc/:$PATH" LC_ALL="" LANG="" dist/raiden-0.2.0+git.r48035549-linux smoketest                         ST 2   master 
Traceback (most recent call last):                                                                                                                                                                                                                                              
  File "raiden-script.py", line 2, in <module>
  File "site-packages/raiden/__main__.py", line 11, in main
  File "site-packages/click/core.py", line 722, in __call__
  File "site-packages/click/core.py", line 676, in main
  File "site-packages/click/_unicodefun.py", line 118, in _verify_python3_env
RuntimeError: Click will abort further execution because Python 3 was configured to use ASCII as encoding for the environment.  Consult http://click.pocoo.org/python3/for mitigation steps.

This system lists a couple of UTF-8 supporting locales that
you can pick from.  The following suitable locales where
discovered: en_US.utf8
[2474] Failed to execute script raiden-script
```

**Comment by konradkonrad** [comment]
When trying to connect with `ropsten`, **after a failed** first attempt (because of incomplete light client data), I get
```
Traceback (most recent call last):
  File "raiden-script.py", line 2, in <module>
  File "site-packages/raiden/__main__.py", line 11, in main
  File "site-packages/click/core.py", line 722, in __call__
  File "site-packages/click/core.py", line 697, in main
  File "site-packages/click/core.py", line 1043, in invoke
  File "site-packages/click/core.py", line 895, in invoke
  File "site-packages/click/core.py", line 535, in invoke
  File "site-packages/click/decorators.py", line 17, in new_func
  File "site-packages/raiden/ui/cli.py", line 707, in run
  File "site-packages/click/core.py", line 535, in invoke
  File "site-packages/raiden/ui/cli.py", line 601, in app
  File "site-packages/raiden/app.py", line 80, in __init__
  File "site-packages/raiden/raiden_service.py", line 248, in __init__
  File "site-packages/raiden/transfer/log.py", line 96, in __init__
  File "site-packages/raiden/transfer/log.py", line 124, in sanity_check
sqlite3.OperationalError: Could not decode to UTF-8 column 'data' with text '�craiden.transfer.state_change
Block
q'
```

`removedb` and restart fixes that issue.

Furtnermore for locale stuff:
```
LC_ALL="" ./raiden-0.2.0+git.r4803554-linux smoketest
# works
LANG="" ./raiden-0.2.0+git.r4803554-linux smoketest
Traceback (most recent call last):
  File "raiden-script.py", line 2, in <module>
  File "site-packages/raiden/__main__.py", line 11, in main
  File "site-packages/click/core.py", line 722, in __call__
  File "site-packages/click/core.py", line 676, in main
  File "site-packages/click/_unicodefun.py", line 118, in _verify_python3_env
RuntimeError: Click will abort further execution because Python 3 was configured to use ASCII as encoding for the environment.  Consult http://click.pocoo.org/python3/for mitigation steps.

This system supports the C.UTF-8 locale which is recommended.
You might be able to resolve your issue by exporting the
following environment variables:

    export LC_ALL=C.UTF-8
    export LANG=C.UTF-8
[6652] Failed to execute script raiden-script
```

**Comment by ulope** [comment]
Hm interesting. We already have a runtime hook in place to detect [missing locale settings](https://github.com/raiden-network/raiden/blob/56d665056e03f8d67ea35a108ccb94086da6feaa/tools/pyinstaller_hooks/runtime_encoding.py#L7-L9) but apparently it's not sufficient on linux.

**Comment by palango** [comment]
@ulope Either it's not enough or click gets executed before the hook is run.

**Comment by ulope** [comment]
@palango No the hook is run by the pyinstaller entrypoint before the "user-code". Also on macos the hook fixes this click error.
Maybe there is a different precedence order in linux compared to macos.

**Comment by palango** [comment]
Fixed by #1248.
